### DISCUSSION TITLE
Changing overvolting and reactors circuits so players actually try to deal electricity mechanics

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#6870] [Ideas and suggestions]
Overvolting tends to destroy all of your junction boxes at the same time (if you keep them repaired). 

This frustrates most players who simply don't try to deal with the mechanic and instead use contraptions such as buffers or worse, Automated Reactor Circuits (ARC) which completely prevent overvolting by being too good

From the steam workshop:
![image](https://user-images.githubusercontent.com/73229309/136622308-23041943-9a20-4c60-89e9-c180251a1fbb.png)

_Reactor controllers are circuits that work as much improved automatic control option for the reactors. Unlike the default automatic control they are insanely fast at reacting to load fluctuations which prevents overvoltages from breaking things and causing fires and they run the reactor at critically low temperature which saves some fuel_

In short, ARCs are extremely powerful, fuel efficient automatic reactors you can install at the start of the campaign.

**Suggested changes:**
1. Remove either the the SET_TURBINEOUTPUT or the SET_FISSIONRATE pin on reactors, or remove both
ARCs rely on both pins. One pin would allow some control over the reactor, but prevent those extremely powerful circuits. We also have reactor PDA on unstable which means you can now use the manual mode from a distance as well, so ARCs become less necessary to handle power fluctuations

2. Change how overvolting damage works.
Instead of always taking damage ticks when overvolted, **make overvolt damage ticks a random chance**
This means that some junction boxes would break faster than other, reducing the overall amount of JBs taking damage by shutting off parts of the circuit. Overvolting wouldn't outright break ALL of your junction boxes right away, making it less frustrating to deal with

![image](https://user-images.githubusercontent.com/73229309/136625475-186a2fca-e041-437d-8eb1-03fce5c07303.png)

**Comment by MateusJK15** [comment]
ARCs are completely optional, there's no one forcing you to use it. Removing it from the game would just upset some players. I use it because manage the reactor is a boring task (it's just 2 sliders going left and right, it's not complex like a real life reactor) and i don't like to see one of my crew members standing still in front of the reactor instead of helping with more important tasks.

**Comment by Ek-Videogames** [comment]
Honestly if the electrical system didn't have a bunch of weird intricate quirks and bugs and where the mechanics spontaneously don't work as intended then balancing a grid load would be alot more fun.
- Get rid of the bugs that are sources of 'unintentional' electrical instability. Phantom electrical problems are not fun.
- Give more solutions, options of ways of managing grid so that you don't have to just resort to battery-buffering anything remotely problematic. It'd be nice to be able to have more fine control over electrical signals and routing, and power/load output pins on junctions to use with said logic.

**Edit:** 
- Main benefit of reactor control is to basically take advantage of the extra 300% of fuel capacity that is otherwise wasted, and also to allow lazy engineers to run all kinds of exceedingly cursed fuel combinations. ( Anything that isn't 2 uranium or 1 thorium / fulgurium )
- If bigger reactors didn't take an unreasonably long time to warm up, and if automatic control could at least semi-correctly run a full 4 rod. Then there would be much less demand for reactor control. ( honestly just make it act like how it does when it has 1 rod, but now with 4 of them lmao problem solved. )

**Reply by MateusJK15** [reply]
Or add an option to upgrade the default reactor control so it become faster with the upgrades.

**Reply by Ek-Videogames** [reply]
- "Fission Capacity" upgrade to allow reactor to run with more rods.

- Faster temperature control upgrade.

**Comment by MateusJK15** [comment]
ARCs are absolutely necessary in this game, because :
Manually control the reactor is EXTREMELY boring, there's only 2 sliders that goes left and right, there's no complexity, there's no challenge, no fun.
If you order a npc to do it, the npc won't allow more than 1 rod inside the reactor, if you put more than 1 there, the npc will just take it out.
The default ARC is so slow that can't reliably manage the reactor if there's more than 1 rod inside it, it keeps overheating the reactor all the time and the game doesn't allow the player to upgrade it at the stations to make it react faster.

**Reply by James-E-A** [reply]
[NB: I agree with the rest of your comment unreservedly, but I can confirm that the integrated ARC *can* cope with multiple rods without overheating if you introduce them properly and carefully. While it'll happily fry junction boxes, I've never seen it sound its alarm bells even once *after initial stabilization* even with 4 rods and a captain doing combat maneuvers.]

**Comment by James-E-A** [comment]
I really like the idea of junction box damage saving later ones down the chain.

If they're damaged, it's because of **heat**, which is dissipated energy. (Hence, the fires.) It would make a *ton* of sense for junction boxes to have a self-Load proportional to their current OV damage rate. This would especially gel with a probabilistic OV damage system to create some more understandable/reasonable/interesting burnout patterns, too.

**Comment by mygamingaccount** [comment]
Overhaul the entire electrical model so that the turbines / batteries are constant voltage sources (as in real life) instead of **constant current sources**.

In this system, problems would arise from junction boxes handling too much current, as that would lead to blown fuses, and overheating conductors catching on fire.

As it is right now, the lagging constant current source turbine can lead to an _overvoltage_ on the grid upon a sudden loss of load.  
With a lagging constant voltage source turbine there will be a _gradual brown out_ on the grid upon a sudden appearance of a great load, until the turbine throttle governor and reactor thermal control catches up. Disappearing load might cause a minor overvoltage situation depending on the reactor throttle response.
